### PR TITLE
Gamified fix logic with isComplete issue #199

### PR DESCRIPTION
**Pull request by loui981c**
## Description

Addressing the problem where only exercise components inside the swiper had isComplete set to true has been resolved. Now, all components are updated, and the swiper starts at the first incomplete component, if any, instead of the start. Additionally, there is a refactoring of the points logic, but the underlying logic remains unchanged.

## Changes

The changes are related to issue #199 

## Related Issues

Issue #199 

## Checklist

- [X] Code has been tested locally and passes all relevant tests.
- [X] Documentation has been updated to reflect the changes, if applicable.
- [X] Code follows the established coding style and guidelines of the project.
- [X] All new and existing tests related to the changes have passed.
- [X] Any necessary dependencies or new packages have been properly documented.
- [X] Pull request title and description are clear and descriptive.
- [X] Reviewers have been assigned to the pull request.
- [X] Any potential security implications have been considered and addressed.
- [X] Performance impact of the changes has been evaluated, if relevant.

## If mobile/frontend pull request, what version of the backend is it stable, and running on? 

Branch: gamified_fix_logic_with_isComplete

Commit id: 4f7a9dc

## Notes for Reviewers

I have commented out some of the tests since the functions do not exist and they can't be found in the current dev branch. Please test with a new user and the third section of the course Finance

### CODE REVIEW

**Comment by benceisonline** [comment]
Tested every functionality on iPhone sim, other than the initial index it looks good 2 me 🦅🦅🦅 Will test again after fix

**Comment by benceisonline** [comment]
Everything works and looking goodie 👍👍👍

File: eml/screens/lectures/CompSwipeScreen.jsx [hunk 1]
```diff
@@ -36,57 +37,20 @@ export default function LectureSwipeScreen({ route }) {
...
-              component: newComp,
-              type: compType,
-              lectureType: lectureType
-            };
-
-            _combinedLecturesAndExercises.push(obj);
-          } catch (error) {
-            console.log(error);
-          }
+        let initialIndex = findIndexOfUncompletedComp(await getStudentInfo(), parsedCourse.courseId, section.sectionId);
```
[the remark this answers is not in the document]
Comment (benceisonline) [reply eml/screens/lectures/CompSwipeScreen.jsx:40]: fixed

**Review by sofiiagran — CHANGES_REQUESTED** [review]
Looks great! However, I would like to not merge it before the dev back-end is up to date. All commits to dev should work with the dev back-end. However, when the back-end changes have been merged to dev, then this PR can be merged :))

**Comment by sofiiagran** [comment]
Also, it looks like you created a direct merge into dev? Is this a mistake? See the photo.. 'https://educado-backend-staging-x7rgvjso4a-ew.a.run.app/'

**Comment by sofiiagran** [comment]
Also see the new merge conflicts

**Review by JackSMunn — APPROVED** [review]
Looks good! :D

**Pull request closed without merging**